dll/shellext/zipfldr/CZipFolder.hpp

```diff
@@ -0,0 +1,627 @@
+
+        DbgPrint("%s\n", __FUNCTION__);
+        return E_NOTIMPL;
+    }
+    STDMETHODIMP MapColumnToSCID(UINT column, SHCOLUMNID *pscid)
+    {
+        UNIMPLEMENTED;
+        return E_NOTIMPL;
+    }
+
+    // *** IShellFolder methods ***
+    STDMETHODIMP ParseDisplayName(HWND hwndOwner, LPBC pbc, LPOLESTR lpszDisplayName, ULONG *pchEaten, PIDLIST_RELATIVE *ppidl, ULONG *pdwAttributes)
+    {
+        UNIMPLEMENTED;
+        return E_NOTIMPL;
+    }
+    STDMETHODIMP EnumObjects(HWND hwndOwner, DWORD dwFlags, LPENUMIDLIST *ppEnumIDList)
+    {
+        return _CEnumZipContents_CreateInstance(this, dwFlags, m_ZipDir, IID_PPV_ARG(IEnumIDList, ppEnumIDList));
+    }
+    STDMETHODIMP BindToObject(PCUIDLIST_RELATIVE pidl, LPBC pbcReserved, REFIID riid, LPVOID *ppvOut)
+    {
+        if (riid == IID_IShellFolder)
+        {
+            CStringA newZipDir = m_ZipDir;
+            PCUIDLIST_RELATIVE curpidl = pidl;
+            while (curpidl->mkid.cb)
+            {
+                const ZipPidlEntry* zipEntry = _ZipFromIL(curpidl);
+                if (!zipEntry)
+                {
+                    return E_FAIL;
+                }
+                newZipDir += zipEntry->Name;
+                newZipDir += '/';
+
+                curpidl = ILGetNext(curpidl);
+            }
+            return ShellObjectCreatorInit<CZipFolder>(m_ZipFile, newZipDir, m_CurDir, pidl, riid, ppvOut);
+        }
+        DbgPrint("%s(%S) UNHANDLED\n", __FUNCTION__, guid2string(riid));
+        return E_NOTIMPL;
+    }
+    STDMETHODIMP BindToStorage(PCUIDLIST_RELATIVE pidl, LPBC pbcReserved, REFIID riid, LPVOID *ppvOut)
+    {
+        UNIMPLEMENTED;
+        return E_NOTIMPL;
+    }
+    STDMETHODIMP CompareIDs(LPARAM lParam, PCUIDLIST_RELATIVE pidl1, PCUIDLIST_RELATIVE pidl2)
+    {
+        const ZipPidlEntry* zipEntry1 = _ZipFromIL(pidl1);
+        const ZipPidlEntry* zipEntry2 = _ZipFromIL(pidl2);
+
+        if (!zipEntry1 || !zipEntry2)
+            return E_INVALIDARG;
+
+        int result = 0;
+        if (zipEntry1->ZipType != zipEntry2->ZipType)
+            result = zipEntry1->ZipType - zipEntry2->ZipType;
+        else
+            result = stricmp(zipEntry1->Name, zipEntry2->Name);
+
+        if (!result && zipEntry1->ZipType == ZIP_PIDL_DIRECTORY)
+        {
+            PCUIDLIST_RELATIVE child1 = ILGetNext(pidl1);
+            PCUIDLIST_RELATIVE child2 = ILGetNext(pidl2);
+
+            if (child1->mkid.cb && child2->mkid.cb)
+                return CompareIDs(lParam, child1, child2);
+            else if (child1->mkid.cb)
+                result = 1;
+            else if (child2->mkid.cb)
+                result = -1;
+        }
+
+        return MAKE_COMPARE_HRESULT(result);
+    }
+    STDMETHODIMP CreateViewObject(HWND hwndOwner, REFIID riid, LPVOID *ppvOut)
+    {
+        static const GUID UnknownIID = // {93F81976-6A0D-42C3-94DD-AA258A155470}
+        {0x93F81976, 0x6A0D, 0x42C3, {0x94, 0xDD, 0xAA, 0x25, 0x8A, 0x15, 0x54, 0x70}};
```

Comment:
There's no existing standard name for this one?

Reply:
Couldn't find it.

Reply:
Apparently not. I did a quick googling and no one has a name for it.